lib/plaid/identity_verification.ex

```diff
@@ -0,0 +1,589 @@
+          kyc_check: Plaid.IdentityVerification.KYCCheck.t() | nil,
+          risk_check: Plaid.IdentityVerification.RiskCheck.t() | nil,
+          watchlist_screening: Plaid.IdentityVerification.WatchlistScreening.t() | nil,
+          request_id: String.t()
+        }
+  @type params :: %{required(atom) => term}
+  @type config :: %{required(atom) => String.t() | keyword}
+  @type error :: {:error, Plaid.Error.t() | any()} | no_return
+
+  defmodule Template do
+    @moduledoc """
+    Plaid Identity Verification Template data structure.
+    """
+
+    @derive Jason.Encoder
+    defstruct id: nil, version: nil
+
+    @type t :: %__MODULE__{
+            id: String.t(),
+            version: integer()
+          }
+  end
+
+  defmodule User do
+    @moduledoc """
+    Plaid Identity Verification User data structure.
+    """
+
+    @derive Jason.Encoder
+    defstruct email_address: nil,
+              phone_number: nil,
+              date_of_birth: nil,
+              ip_address: nil,
+              name: nil,
+              address: nil,
+              id_number: nil
+
+    @type t :: %__MODULE__{
+            email_address: String.t() | nil,
+            phone_number: String.t() | nil,
+            date_of_birth: String.t() | nil,
+            ip_address: String.t() | nil,
+            name: Plaid.IdentityVerification.User.Name.t() | nil,
+            address: Plaid.IdentityVerification.User.Address.t() | nil,
+            id_number: Plaid.IdentityVerification.User.IDNumber.t() | nil
+          }
+
+    defmodule Name do
+      @moduledoc """
+      Plaid Identity Verification User Name data structure.
+      """
+
+      @derive Jason.Encoder
+      defstruct given_name: nil, family_name: nil
+
+      @type t :: %__MODULE__{
+              given_name: String.t(),
+              family_name: String.t()
+            }
+    end
+
+    defmodule Address do
+      @moduledoc """
+      Plaid Identity Verification User Address data structure.
+      """
+
+      @derive Jason.Encoder
+      defstruct street: nil,
+                street2: nil,
+                city: nil,
+                region: nil,
+                postal_code: nil,
+                country: nil
+
+      @type t :: %__MODULE__{
+              street: String.t(),
+              street2: String.t() | nil,
+              city: String.t(),
+              region: String.t(),
+              postal_code: String.t(),
+              country: String.t()
+            }
+    end
+
+    defmodule IDNumber do
+      @moduledoc """
+      Plaid Identity Verification User ID Number data structure.
+      """
+
+      @derive Jason.Encoder
+      defstruct value: nil, type: nil
+
+      @type t :: %__MODULE__{
+              value: String.t(),
+              type: String.t()
+            }
+    end
+  end
+
+  defmodule Steps do
+    @moduledoc """
+    Plaid Identity Verification Steps data structure.
+
+    Each step will be one of the following values: `active`, `success`, `failed`,
```

Comment:
FMI, which field in the struct has this value? Asking to see whether it makes sense to make them explicit `@type` values, perhaps as atoms